### PR TITLE
Move to a peer dependency for typescript

### PR DESCRIPTION
**Pull request by JKillian**
Accept all versions of 1.7.0-dev after the jsx spread bugfix, accept all versions of 1.8.0-dev, accept any release versions >= 1.7.0

### CODE REVIEW

**Comment by JKillian** [comment]
Addresses #791

**Comment by JKillian** [comment]
hmm, this only seems to fail on Node 0.10 with a 'typescript' not found warning. Any ideas @adidahiya?

File: package.json [hunk 1]
```diff
@@ -38,5 +37,8 @@
     "mocha": "^2.2.5",
     "tslint": "latest"
```
Comment (adidahiya) [inline package.json]: I think you might need to keep typescript as a dev dependency

**Comment by JKillian** [comment]
Might be because with Node 0.10 it uses npm 1.x whereas with Node 0.12+ it uses npm 2.x

File: package.json [hunk 2]
```diff
@@ -36,7 +35,11 @@
     "grunt-ts": "^5.1.0",
     "grunt-tslint": "latest",
     "mocha": "^2.2.5",
-    "tslint": "latest"
+    "tslint": "latest",
+    "typescript": ">=1.7.0 || >=1.7.0-dev.20151003 || >=1.8.0-dev"
```
Comment (adidahiya) [inline package.json]: I think the dev dependency should be `next`, like it was before

Comment (JKillian) [inline package.json]: Good call, just pushed an update with that change

**Pull request merged**